corehq/apps/cloudcare/static/cloudcare/js/formplayer/menus/views/query.js

```diff
@@ -728,23 +728,46 @@ hqDefine("cloudcare/js/formplayer/menus/views/query", [
         submitAction: function (e) {
             var self = this;
             e.preventDefault();
-            self.performSubmit();
+            self.performSubmit(null, formplayerConstants.USER_CLICK_SUBMIT);
         },
 
-        performSubmit: function (initiatedBy) {
+        performSubmit: function (initiatedBy, userClickSubmit = null) {
             var self = this;
-            self.validateAllFields().done(function () {
-                FormplayerFrontend.trigger(
-                    "menu:query",
-                    self.getAnswers(),
-                    self.options.sidebarEnabled,
-                    initiatedBy
-                );
-                if (self.smallScreenEnabled && self.options.sidebarEnabled) {
-                    $('#sidebar-region').collapse('hide');
+            if (!self.inputsHaveErrors()) {
+                self.executeSearch(initiatedBy).done(function (response) {
+                    self.updateModels(response);
+                });
+            }
+            if (userClickSubmit === formplayerConstants.USER_CLICK_SUBMIT) {
+                self.displayErrors();
+            }
```

Comment:
since the body of this is only one line, and it's only used once, what would you think about just moving `self.displayErrors();` into `submitAction` and dropping the `userClickSubmit` arg?

Reply:
changed here [8be4ffa](https://github.com/dimagi/commcare-hq/pull/34658/commits/8be4ffafd2470a16b68858ba49777d81ce57512a)